app/database/operator/server_data_operator/comparators/index.ts

```diff
@@ -33,6 +35,14 @@ import type UserModel from '@typings/database/models/servers/user';
  *  'record' and the 'raw'
  */
 
+export const isRecordCategoryEqualToRaw = (record: CategoryModel, raw: Category) => {
+    return raw.id === record.id;
+};
+
+export const isRecordCategoryChannelEqualToRaw = (record: CategoryChannelModel, raw: CategoryChannel) => {
+    return (record.id === raw.id);
```

Comment:
don't you need to check for `record.categoryId === raw.category_id && record.channelId === raw.channel_id` as well? it looks to me that you do want to check that both are equal to determine that it is the same element cause the `id` for the ternary is generated locally, right ?

Reply:
Almost - the `categoryChannel.id` is generated locally by `${team_id}_${channel_id}` so we can just compare on ID; 

https://github.com/mattermost/mattermost-mobile/pull/5874/files#diff-985245b091be0a8a63434f2cf4c63a79ff4cb8f814e3c0735f07653f6be9a919R106

Reply:
this should not be part of another PR but you should construct the `id` in the handler instead or something similar, is to easy to omit in case is needed to be set in other places

Reply:
probably the best place could be the transformer

Reply:
Yup that's what I'm thinking - will test!

Reply:
I think it will have to live outside - the `team_id` isn't available in the handler/transformer on this object without querying for it; which is probably outside of the handler/transformers scope. The ID is currently set in the `prepareCategoryChannels` method, which is probably the best place for it.

I can't really use any other field either as `team_id` and `channel_id` are the only two constants (`sort_order` and `category_id` are changeable, and `user_id` with `channel_id` will result in dupes for DM/GMs).

Reply:
> Can the id be just autogenerated by watermelon instead of having it composed by some vslues that may or not may be available at the same time in a specific scenario?

I did it that way first, but it meant that for each `CategoryChannel` item I had to see if I could find it using the `team_id` + `channel_id` + `user_id` and then do an update, so not the most efficient. I then tried to just do a deleteAll (for that user/team) and insert new (just two queries), but that was also not as fast as just generating the id (using team/channel id) and just doing an update (one write). With async db batching, and multiple websocket update events coming through; I wasn't sure if there would also be race condition issues there.

Reply:
I guess is fine, if we end up having other issues we can revisit